### PR TITLE
Collections Dev Guide Update

### PR DESCRIPTION
**Pull request by domenicbove**
# Description

- [x] Update DEVELOPMENT_GUIDE.md with the new role paths and filters changed with the collections work
- [x] Update HOW_TO_TEST.md with new paths for molecule scenarios
- [x] Add git clone instructions for development. If you git clone without the ansible collections directory structure, then molecule fails to find plugins. This needs to be added somewhere:
```
mkdir -p ansible_collections/confluent/platform
git clone https://github.com/confluentinc/cp-ansible.git ansible_collections/confluent/platform
```

Relates to # (https://github.com/confluentinc/cp-ansible/issues/613)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Just a docs update, needs a read thru by reviewer

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] Any variable changes have been validated to be backwards compatible

### CODE REVIEW

File: DEVELOPMENT_GUIDE.md [hunk 1]
```diff
@@ -176,29 +176,31 @@ Because properties are fully customizable, it is important to use the final prop
 
 ## Custom Filters
 
-Ansible itself has a robust set of filters, but at times they do not fit cp-ansible’s needs. We have defined additional filters at `plugins/filter/filters.py`. In the below example we combine a custom filter `get_sasl_mechanisms` and our of the box ansible filters to set a variable:
+Ansible itself has a robust set of filters, but at times they do not fit cp-ansible’s needs. We have defined additional filters at `plugins/filter/filters.py`. In the below example we combine a custom filter `get_sasl_mechanisms` and one of the box ansible filters to set a variable:
```
Comment (erikgb) [inline DEVELOPMENT_GUIDE.md:179]:
```suggestion
Ansible itself has a robust set of filters, but at times they do not fit cp-ansible’s needs. We have defined additional filters at `plugins/filter/filters.py`. In the below example we combine a custom filter `get_sasl_mechanisms` and one of the standard ansible filters to set a variable:
```

File: DEVELOPMENT_GUIDE.md [hunk 2]
```diff
@@ -176,29 +176,31 @@ Because properties are fully customizable, it is important to use the final prop
...
 
 ```
 kafka_broker_sasl_enabled_mechanisms: "{{ kafka_broker_listeners | confluent.platform.get_sasl_mechanisms(sasl_protocol) | difference(['none']) | unique }}"
 ```
 
 The `kafka_broker_sasl_enabled_mechanisms` variable is a list built out of all of the sasl mechanisms defined in the `kafka_broker_listeners` dictionary.
 
+Cp-ansible is written as an Ansible Collection. This means all custom filter invokations must use the Fully Qualified Filter Name, or simply put, must begin with `confluent.platform.`
```
Comment (erikgb) [inline DEVELOPMENT_GUIDE.md:187]:
```suggestion
Cp-ansible is written as an Ansible Collection. This means all custom filter invocations must use the Fully Qualified Filter Name, or simply put, must begin with `confluent.platform.`
```

**Review by erikgb — APPROVED** [review]
LGTM, but we might consider to keep this PR open until we are "almost" done. There might be other updates to the dev guide....

**Comment by domenicbove** [comment]
@erikgb good point! I need to add these dev steps somewhere 
```
mkdir -p ansible_collections/confluent/platform
git clone https://github.com/confluentinc/cp-ansible.git ansible_collections/confluent/platform
```
without this molecule wont work!

File: DEVELOPMENT_GUIDE.md [hunk 3]
```diff
@@ -176,37 +176,41 @@ Because properties are fully customizable, it is important to use the final prop
...
-Ansible itself has a robust set of filters, but at times they do not fit cp-ansible’s needs. We have defined additional filters at `plugins/filter/filters.py`. In the below example we combine a custom filter `get_sasl_mechanisms` and our of the box ansible filters to set a variable:
+Ansible itself has a robust set of filters, but at times they do not fit cp-ansible’s needs. We have defined additional filters at `plugins/filter/filters.py`. In the below example we combine a custom filter `get_sasl_mechanisms` and one of the standard ansible filters to set a variable:
 
 ```
 kafka_broker_sasl_enabled_mechanisms: "{{ kafka_broker_listeners | confluent.platform.get_sasl_mechanisms(sasl_protocol) | difference(['none']) | unique }}"
 ```
 
 The `kafka_broker_sasl_enabled_mechanisms` variable is a list built out of all of the sasl mechanisms defined in the `kafka_broker_listeners` dictionary.
 
+Cp-ansible is written as an Ansible Collection. This means all custom filter invocations must use the Fully Qualified Filter Name, or simply put, must begin with `confluent.platform.`
```
Comment (domenicbove) [inline DEVELOPMENT_GUIDE.md:187]:
```suggestion
Cp-ansible is written as an Ansible Collection. This means all custom filter invocations must use the Fully Qualified Filter Name, or simply put, must be prefixed with `confluent.platform.`
```

File: DEVELOPMENT_GUIDE.md [hunk 4]
```diff
@@ -1,6 +1,6 @@
 # Development Guide
 
-The following document will be a reference guide to coding standards of cp-ansible. It is written to be idempotent, meaning whatever you set in your inventory file, the playbook and roles should provision on your hosts, and can be run multiple times without causing unnecessary changes to those hosts.
+The following document will be a reference guide to coding standards of cp-ansible. It is written to be idempotent, meaning whatever you set in your inventory file, the playbooks and roles should provision on your hosts, and can be run multiple times without causing unnecessary changes on those hosts or restarts to those services.
```
Comment (erikgb) [inline DEVELOPMENT_GUIDE.md:3]:
```suggestion
The following document will be a reference guide to coding standards of cp-ansible. It is written to be idempotent, meaning whatever you set in your inventory file, the playbooks and roles should provision on your hosts, and can be run multiple times without causing unnecessary changes nor service restarts on those hosts.
```

File: HOW_TO_TEST.md [hunk 5]
```diff
@@ -1,97 +1,115 @@
...
 ```
-ImportError: No module named docker.common
+python3 -m pip install --user ansible
 ```
-
-As a workaround you can use molecule in a container.  
-In your current shell create an alias to start molecule in a container:
-
+2. [Docker](https://docs.docker.com/get-docker/) *Note: We recommend increasing your docker memory to at least 20GB of RAM and your CPU count to 10.*
+3. [Molecule](https://molecule.readthedocs.io/en/latest/installation.html#install). Use python3's pip for installation ie
```
Comment (erikgb) [inline HOW_TO_TEST.md:13]: I think we have a requirement on the Molecule version. At least 3.x.

Comment (domenicbove) [reply HOW_TO_TEST.md:13]: nice catch

File: HOW_TO_TEST.md [hunk 6]
```diff
@@ -1,97 +1,115 @@
...
-Note: If using Molecule version 3.1.0 or later use the following command to install:
-
-```pip3 install molecule molecule-docker```
-
-
-### Running Molecule in a Container
-
-At times, pip installation of molecule can lead to errors:
+1. Python3
+2. [Ansible >= 2.11 installed with Pip](https://docs.ansible.com/ansible/latest/installation_guide/intro_installation.html#installing-ansible-with-pip). Use python3's pip for ansible installation to make sure ansible is not configured with Python2.7:
```
Comment (domenicbove) [inline HOW_TO_TEST.md:8]:
```suggestion
2. [Ansible >= 2.11](https://docs.ansible.com/ansible/latest/installation_guide/intro_installation.html#installing-ansible-with-pip). Use python3's pip for ansible installation to make sure ansible is not configured with Python2.7:
```

File: HOW_TO_TEST.md (hunk 5 above)
Comment (domenicbove) [reply HOW_TO_TEST.md:13]:
```suggestion
3. [Molecule >= 3.3](https://molecule.readthedocs.io/en/latest/installation.html#install). Use python3's pip for installation: 
```

File: HOW_TO_TEST.md [hunk 7]
```diff
@@ -1,97 +1,115 @@
...
 
-To make the alias permanent, add the following to your bashrc file:
+## Cloning CP-Ansible
+
+CP-Ansible is written as an [Ansible Collection](https://docs.ansible.com/ansible/latest/dev_guide/developing_collections.html), which requires a rigid directory structure for development. In order for Molecule to succeed in finding custom filters and plugins, cp-ansible must be cloned into an `ansible_collections/` directory. If you already have cp-ansible cloned run:
 
 ```
-export CP_ANSIBLE_PATH=<Replace this with the repo path>
-alias molecule="docker run -it --rm --dns="8.8.8.8" -v "/var/run/docker.sock:/var/run/docker.sock" -v ~/.cache:/root/.cache -v "$CP_ANSIBLE_PATH:$CP_ANSIBLE_PATH" -w "$CP_ANSIBLE_PATH/roles/confluent.test" quay.io/ansible/molecule:3.1.5 molecule"
+mkdir -p ansible_collections/confluent
```
Comment (erikgb) [inline HOW_TO_TEST.md:23]: Do we really need this? Can't we just make it simple, and suggest the approach below?

`git clone https://github.com/confluentinc/cp-ansible.git ansible_collections/confluent/platform`

Comment (domenicbove) [reply HOW_TO_TEST.md:23]: good question, i assumed that would fail if the in between dirs did not exist, ill test

Comment (domenicbove) [reply HOW_TO_TEST.md:23]: works great, nice suggestion

Comment (erikgb) [reply HOW_TO_TEST.md:23]: If someone wants to move their existing clone, they probably know how to do it. 😄

**Review by erikgb — APPROVED** [review]
LGTM

**Pull request merged**